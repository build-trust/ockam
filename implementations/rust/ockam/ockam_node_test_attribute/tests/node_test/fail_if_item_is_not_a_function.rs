// This test checks that #[ockam_node_test_attribute::node_test] causes a compile time error
// if the item it is defined on is not a function.

#[ockam_node_test_attribute::node_test]
struct A {}
