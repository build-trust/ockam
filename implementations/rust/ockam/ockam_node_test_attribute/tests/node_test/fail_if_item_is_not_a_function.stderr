error: expected `fn`
 --> $DIR/fail_if_item_is_not_a_function.rs:5:1
  |
5 | struct A {}
  | ^^^^^^

error[E0601]: `main` function not found in crate `$CRATE`
 --> $DIR/fail_if_item_is_not_a_function.rs:4:1
  |
4 | / #[ockam_node_test_attribute::node_test]
5 | | struct A {}
  | |___________^ consider adding a `main` function to `$DIR/tests/node_test/fail_if_item_is_not_a_function.rs`
